### PR TITLE
feat: standardize formatting and linting setup

### PR DESCRIPTION
**Pull request by Jacksonmills**
Enhanced the code quality tools by integrating Prettier for consistent code formatting, along with new eslint configurations and plugins to ensure adherence to best practices. Specifically, the changes extend eslint rules for Next.js and TailwindCSS, ignore certain file patterns, and enforce stylistic consistency. The eslint-plugin-tailwindcss and eslint-plugin-react improvements aim to smooth out the development experience.

This commit also adds .prettierignore and prettier.config.cjs to manage files Prettier should ignore and to specify the code style rules, respectively. Lastly, the eslintignore and eslintrc.json files were created and updated, anchoring the linting process closer to the project's requirements.

### CODE REVIEW

**Comment by Jacksonmills** [comment]
Not sure why but prettier was not happy unless I moved it to the root... works fine afaik and we want it global for the other .ts files I guess but will need backend people to test for me

File: app/package.json [hunk 1]
```diff
@@ -7,6 +7,11 @@
     "build": "next build",
     "start": "next start",
     "lint": "next lint",
+    "lint:fix": "next lint --fix",
+    "format:check": "prettier --check .",
+    "format": "prettier --write . --list-different",
+    "check": "npm run lint && npm run format:check",
+    "fix": "npm run lint:fix && npm run format",
```
Comment (Jacksonmills) [inline app/package.json:14]: Added these scripts I have found really helpful, basic use is you can just run `npm run check` and then `fix` if needed before committing and then we can assure consistent formatting/linting

Comment (Jacksonmills) [reply app/package.json:14]: @adamcohenhillel

**Review by adamcohenhillel — APPROVED** [review]

**Comment by adamcohenhillel** [comment]
Looks good to me

**Pull request merged**